src/lib.rs

```diff
@@ -0,0 +1,124 @@
+
+//! ## Start-up sequence summary:
+//! - QEMU loads hypervisor kernel (this program) and linux kernel (held in initrd) into memory
+//! - QEMU launches hardcoded mrom reset vector, which jumps to 0x80000000
+//! - _start is located at 0x80000000 as the only function in the .init.entrypoint section
+//! - `_start` sets up the stack and calls into mstart
+//! - `mstart` initializes machine-mode control registers as needed by the hypervisor
+//! - `mstart` returns into supervisor-mode in sstart
+//! - `sstart` returns into user-mode at the guest kernel entrypoint
+//!       (running in emulated-supervisor-mode)
```

Comment:
This comment is now out-of-date; perhaps either update it or remove it?

Reply:
Good catch